src/Microsoft.DotNet.Tools.Builder/Program.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using NuGet.Frameworks;
+
+
+namespace Microsoft.DotNet.Tools.Build
+{
+    public class Program
+    {
+        public static int Main(string[] args)
+        {
+            DebugHelper.HandleDebugSwitch(ref args);
+
+            try
+            {
+                var app = new CompilerCommandApp("dotnet build", ".NET Builder", "Builder for the .NET Platform. It performs incremental compilation if it's safe to do so. Otherwise it delegates to dotnet-compile which performs non-incremental compilation");
+                return app.Execute(OnExecute, args);
+            }
+            catch (Exception ex)
+            {
+#if DEBUG
+                Console.Error.WriteLine(ex);
+#else
+                Console.Error.WriteLine(ex.Message);
+#endif
+                return 1;
+            }
+        }
+
+        private static bool OnExecute(List<ProjectContext> contexts, string configValue, string outputValue, string intermediateValue, bool noHost,
+            bool isNative, string archValue, string ilcArgsValue, string ilcPathValue, string ilcSdkPathValue,
+            bool isCppMode)
+        {
+            foreach (var context in contexts)
+            {
+                // Set up Output Paths
+                string outputPath = context.GetOutputPath(configValue, outputValue);
+                string intermediateOutputPath = context.GetIntermediateOutputPath(configValue, intermediateValue, outputValue);
+
+                Directory.CreateDirectory(outputPath);
+                Directory.CreateDirectory(intermediateOutputPath);
```

Comment:
Not a comment on this commit, so no need to fix anything since I know this is a direct copy from Compile. 
But CreateDirectory can throw IOException and UnauthorizedAccess exception. Not sure what the patter for error handling in the CLI is here. Just blow up and hope the original exception has enough info, or should we provide custom handling?